tftypes/primitive.go

```diff
@@ -102,30 +102,11 @@ func (p primitive) supportedGoTypes() []string {
 	case Bool.name:
 		return []string{"bool", "*bool"}
 	case DynamicPseudoType.name:
-		possibleTypes := []Type{
-			String, Bool, Number,
-			Tuple{}, Object{},
-		}
-		results := []string{}
-		for _, t := range possibleTypes {
-			results = append(results, t.supportedGoTypes()...)
-		}
-		return results
+		return []string{"nil"}
```

Comment:
Wouldn't we want `UnknownValue` here, too?

Reply:
Indeed! Will fix.

```suggestion
		return []string{"nil", "UnknownValue"}
```